packages/@react-aria/selection/src/useSelectableItem.ts

```diff
@@ -234,20 +258,31 @@ export function useSelectableItem(options: SelectableItemOptions): SelectableIte
     }
   });
 
-  // Pressing the Enter key with selectionBehavior = 'replace' performs an action (i.e. navigation).
-  let onKeyUp = hasSecondaryAction ? (e: KeyboardEvent) => {
-    if (e.key === 'Enter') {
-      onAction();
+  // Prevent native drag and drop on long press if we also select on long press.
+  // Once the user is in selection mode, they can long press again to drag.
+  let onDragStart = e => {
+    if (modality.current === 'touch' && longPressEnabledOnPressStart.current) {
+      e.preventDefault();
     }
-  } : undefined;
+  };
 
   return {
     itemProps: mergeProps(
       itemProps,
       allowsSelection || hasPrimaryAction ? pressProps : {},
-      hasSecondaryAction ? longPressProps : {},
-      {onKeyUp, onDoubleClick}
+      longPressEnabled ? longPressProps : {},
+      {onDoubleClick, onDragStart}
     ),
     isPressed
   };
 }
+
+function isActionKey() {
+  let event = window.event as KeyboardEvent;
+  return event?.key === 'Enter';
+}
+
+function isSelectionKey() {
+  let event = window.event as KeyboardEvent;
```

Comment:
deprecated https://developer.mozilla.org/en-US/docs/Web/API/Window/event
should we add more fields to what we pass through the press events?

Reply:
It works in all browsers and will likely never be removed because it would break all sorts of sites.